### PR TITLE
[DEV-3617] Add /filter/environments

### PR DESCRIPTION
**Pull request by vmarkovtsev**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -639,6 +639,51 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GenericError'
+          description: The specified account does not exist or the user is not a member.
+        422:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSourceDataError'
+          description: The account hasn't submitted any deployment notifications.
```
Comment (warenlg) [inline openapi.yaml:678]: So you differentiate no envs ever and no envs for the given date range, indeed, that's good!

File: openapi.yaml [hunk 2]
```diff
@@ -10142,3 +10187,65 @@ components:
...
+      type: array
+    FilteredEnvironment:
+      additionalProperties: false
+      description: Details about a deployment environment.
+      properties: 
+        name:
+          description: Name of the environment.
+          type: string
+        deployments_count:
+          description: Number of deployments (successful or not) in the specified time interval.
```
Comment (warenlg) [inline openapi.yaml:10238]: This might be useful indeed.

File: openapi.yaml [hunk 3]
```diff
@@ -10142,3 +10187,65 @@ components:
...
+          minimum: 1
+          type: integer
+        last_conclusion:
```
Comment (warenlg) [inline openapi.yaml:10241]: This is not necessary

**Review by warenlg — COMMENTED** [review]
Imaging we implement a dropdown in the UI one day which should list all envs ever independently from the filter selection. What query can return those?

File: openapi.yaml (hunk 3 above)
Comment (vmarkovtsev) [reply openapi.yaml:10241]: I followed GitHub commit statuses here. They show either a green check mark or a red cross for the latest commit in the repository.

**Comment by vmarkovtsev** [comment]
To list all envs ever you set date_from to 2010 and date_to to today. Should be fine. I don't want to make dates null able, that's not consistent with the rest of the endpoints.

**Comment by warenlg** [comment]
That's what I thought as well.

File: openapi.yaml (hunk 3 above)
Comment (warenlg) [reply openapi.yaml:10241]: That's interesting, we won't use it in the MVP, but get your point, will likely be useful as an enhancement.

**Review by warenlg — APPROVED** [review]

**Pull request merged**